src/views/forms/TradeForm/TradeFormInputs.tsx

```diff
@@ -41,6 +43,19 @@ type TradeBoxInputConfig = {
   slotRight?: React.ReactNode;
 };
 
+// a reference to source that only updates once gate is true
+function useGatedValue<T>(source: T, gate: boolean): T | undefined {
+  const [value, setValue] = useState<T>();
+  useEffect(() => {
+    if (value !== source && gate) {
+      setValue(source);
+    }
+    // don't depend on the thing we change
```

Comment:
i think this is fine since on the next render value == source and it'll exit 

i think if you dont pass in `value` to the dep array here you risk not comparing to the right value since its not included in the closure? only happens if you can setValue from another place though i guess